testtools/tests/test_deferredruntest.py

```diff
@@ -620,11 +622,17 @@ def test_something(self):
         result = self.make_result()
         runner.run(result)
         self.assertThat(
-            result._events,
+            [event[:2] for event in result._events],
             Equals([
                 ('startTest', test),
-                ('addSuccess', test, {'twisted-log': text_content('')}),
+                ('addSuccess', test),
                 ('stopTest', test)]))
+        error = result._events[1][2]
+        self.assertThat(error, KeysEqual('twisted-log'))
+        self.assertThat(
+            error['twisted-log'].as_text(),
+            ContainsAll(
+                ['Traceback (most recent call last):', 'ZeroDivisionError']))
```

Comment:
I think this might be easier to read if we had a helper that materialised the content here, giving us a simple basic data type structure we can compare to - because there is no dynamic content in the test.